ztp/siteconfig-generator/main.go

```diff
@@ -27,13 +27,13 @@ func main() {
 	for _, siteConfigFile := range siteConfigFiles {
 		fileData, err := siteConfigs.ReadFile(siteConfigFile)
 		if err != nil {
-			log.Fatalf("Error: could not read file %s: %s\n", siteConfigFile, err)
+			log.Printf("Error: could not read file %s: %s\n", siteConfigFile, err)
```

Comment:
> What are the side-effects of converting the Fatalf to Printf?
> 
> * Does the plugin continue to generate CRs for other SiteConfigs that were modified in the same commit?
Yes, it will continue to generate the other siteconfig CRs
> * Is the user able to see the error somewhere?
Yes, in the terminal output there is an error message show up. In ArgoCD App, All the siteConfig CRs will be missing OR  some of them based on the mistake.

> * Does the user get any feedback that there was an error in generating artifacts for this SiteConfig?
>  yes, simple as the expected managedCluster  CR is missing. 
>   * Can the error be propagated down to the hub cluster, for example by adding an error annotation to the generated namespace?
why adding annotation to the NS ? I don't think this a good idea.

Reply:
The idea of adding an annotation is to help surface the issue to the user. Is my understanding correct that, if the user has a SiteConfig with an error in git, with this change they will still get a green check-mark in argocd with "Sync OK"? If so, how do we make the error obvious/seen by the user if they start looking at the hub cluster and notice that the ManagedCluster is missing? It doesn't have to be an annotation, just looking to make it easier for the user.

Reply:
As we discussed, siteConfig CR will be written if there is error in the siteConfig structure.